fortanix-sgx-tools/src/bin/ftxsgx-runner.rs

```diff
@@ -7,38 +7,61 @@
 extern crate aesm_client;
 extern crate enclave_runner;
 extern crate sgxs_loaders;
-#[macro_use]
 extern crate failure;
-
-use std::path::Path;
+#[macro_use]
+extern crate clap;
 
 use aesm_client::AesmClient;
-use enclave_runner::Command;
+use enclave_runner::EnclaveBuilder;
 use failure::{Error, ResultExt};
 use sgxs_loaders::isgx::Device as IsgxDevice;
 
+use clap::{App, Arg};
+
+arg_enum!{
+    #[derive(PartialEq, Debug)]
+    #[allow(non_camel_case_types)]
+    pub enum Signature {
+        coresident,
+        dummy
+    }
+}
+
 fn main() -> Result<(), Error> {
-    let mut args = std::env::args_os();
-    let cmdname = args.next();
-    let file = match args.next() {
-        Some(arg) => arg,
-        None => {
-            let cmdname = cmdname
-                .as_ref()
-                .and_then(|p| Path::new(p).file_name())
-                .map(|p| Path::new(p).display().to_string());
-            eprintln!(
-                "Usage: {} <ENCLAVE>\n",
-                cmdname.unwrap_or("ftxsgx-runner".into())
-            ); // extra newline
-            bail!("Missing <ENCLAVE> parameter on command line");
-        }
-    };
+    let default_signature_value = Signature::coresident.to_string();
+
+    let args = App::new("ftxsgx-runner")
+        .arg(
+            Arg::with_name("file")
+                .required(true)
+        )
+        .arg(Arg::with_name("signature")
+            .short("s")
+            .long("signature")
+            .required(false)
+            .takes_value(true)
+            .possible_values(&Signature::variants())
+            .default_value(&default_signature_value))
+        .get_matches();
+
+    let file = args.value_of("file").unwrap();
+    let signature = args.value_of("signature").unwrap();
+
     let mut device = IsgxDevice::new()
         .context("While opening SGX device")?
         .einittoken_provider(AesmClient::new())
         .build();
-    let enclave = Command::new(&file, &mut device).context("While loading SGX enclave")?;
+
+    let mut enclave_builder = EnclaveBuilder::new(file.as_ref());
+
+    match signature {
+        "dummy"=> { enclave_builder.dummy_signature(); },
+        "coresident" => { enclave_builder.coresident_signature().context("While loading coresident signature")?; },
```

Comment:
Ok but now I can't get the default behavior. Maybe make the argument optional (instead of having a default)? You can so something like this:
```rust
match args.value_of("signature").map(|v| v.parse().unwrap()) {
    Some(Signature::coresident) => ...,
    Some(Signature::dummy) => ...,
    None => ...,
}
```

Reply:
Done